code/game/objects/items/devices/traitordevices.dm

```diff
@@ -262,3 +262,24 @@ effective or pretty fucking useless.
 	else
 		GLOB.active_jammers -= src
 	update_icon()
+
+/obj/item/storage/toolbox/emergency/turret
+	desc = "You feel the need to hit this one with a wrench"
+
+/obj/item/storage/toolbox/emergency/turret/PopulateContents()
+	new /obj/item/screwdriver(src)
+	new /obj/item/wrench(src)
+	new /obj/item/weldingtool(src)
+	new /obj/item/crowbar(src)
+	new /obj/item/analyzer(src)
+	new /obj/item/wirecutters(src)
+
+/obj/item/storage/toolbox/emergency/turret/attackby(obj/item/I, mob/living/user, params)
+	..()
+
+	if(I.tool_behaviour == TOOL_WRENCH && user.a_intent == INTENT_HARM)
+		user.visible_message("<span class='danger'>[user] bashes [src] with [I]!</span>", \
+			"<span class='danger'>You bash [src] with [I]!</span>", null, COMBAT_MESSAGE_RANGE)
+		playsound(src, "sound/items/drill_use.ogg", 80, TRUE, -1)
+		new /obj/machinery/porta_turret/syndicate/pod/toolbox(get_turf(src.loc))
```

Comment:
```suggestion
		var/obj/machinery/porta_turret/syndicate/pod/toolbox/turret
		turret.creator = user
		new turret(get_turf(src.loc))
```

Reply:
Is this how it works? Wouldn't `turret.creator = user` error since you're trying to set `creator` of null? Wouldn't this make more sense as a parameter to Initialize?

Reply:
yeah probably should be an initialize parameter but im too lazy to update the code suggestion

Reply:
hell you probably dont even need any code on the turret itself and just edit the faction in this proc